public/scripts/experiments/jnd.js

```diff
@@ -606,19 +617,57 @@ class JND {
       var xAxisElements = chart.append("g")
                                 .attr("transform", "translate(50, " + xAxisTranslate  +")")
                                 .call(x_axis)
-
-      // Populating data: 
-      chart.selectAll("circle") // Technically no circles inside div yet, but will be creating it
-           .data(data)
-              .enter()
-              .append("circle") // Creating the circles for each entry in data set 
-              .attr("cx", function (d) { // d is a subarray of the dataset i.e coordinates [5, 20]
-                return xscale(d[0]) + 60; // +60 is for buffer (points going -x, even if they are positive)
-              })
-              .attr("cy", function (d) {
-                return yscale(d[1]);
-              })
-              .attr("r", trial_data.point_size).style("fill", trial_data.point_color);
+           
+      // TODO: Different handling for distractor - needs to be abstracted out somehow in future     
+      if (this.condition_name.split("_")[0] === "distractor"){ 
+           
+        let dataset = datasets[i];
+        let distractor = distractors[i];
+
+        // Alternate plotting of distractor and main dataset points - want equal chance of one
+        // getting occluded over the other
+        for (let j in dataset) {
+
+          let point = dataset[j];
+          let dist_point = distractor[j];
+
+          chart.selectAll("circle_data")
+               .data([point])
+                .enter()
+                .append("circle") // Creating the circles for each entry in data set 
+                .attr("cx", function (d) { // d is a subarray of the dataset i.e coordinates [5, 20]
+                  return xscale(d[0]) + 60; // +60 is for buffer (points going -x, even if they are positive)
+                })
+                .attr("cy", function (d) {
+                  return yscale(d[1]);
+                })
+                .attr("r", trial_data.point_size).style("fill", trial_data.point_color);
+
+          chart.selectAll("circle_data")
```

Comment:
lots of stuff that probably could be outside this if/else

if (...) {
string chartId = "foo";
int[][] dataset = [[1,2], [2,4]];
// set some other variables
} else { ... }

chart.selectAll(chartId)
.data(dataset)
.enter
.append("circle")

(...)